README.md

```diff
@@ -1,11 +1,165 @@
 # `apropos-tx`
+Generate a suite of property tests from a logical model.
 
 ## What is this?
+It's a thing that generates property tests for you from a model.
+
+### What kind of model?
+A logical model that describes the device you are testing.
+
+### What device?
+This is the thing you are hooking your test equipment up to.
+
+### What?
+It's a function.
+
+### And how do you write this logical model?
+Good question. First you start with a model. Then you describe it's properties.
+
+### What's a model?
+It's any datatype of your choosing... For example it could be this thing
+
+```Haskell
+data MyModel =
+  This | That | TheOther { things :: [Thing] }
+```
+
+### And the properties?
+They are an enum like this.
+
+```Haskell
+data MyProperty =
+    DoesThis
+  | DoesThat
+  | DoesTheNextThing
+  deriving (Bounded,Eq,Ord,Enum,Show)
+
+```
+
+It satisfies the constraints required for these terms to be used as Propositions in a Propositional Calculus.
+
+This means we can write logical statements in terms of them.
+
+For Example.
+
+```Haskell
+term = (Var DoesThis :->: Var DoesThat)
+    :&&: (Var DoesThat :->: Var DoesTheNextThing)
+
+```
+
+Neat! Terms like these are used to encode Three aspects of our model.
+1. Model Internal Logic
+2. Expected Device Response
+3. Generator Transformation Matches
+
+#### Model Internal Logic
+This is a term you define to constrain which properties of the model can be true at the same time.
+
+```Haskell
+instance LogicalModel MyProperty where
+  logic = Var This :->: Var That
+```
+
+For example if This implies That then whenever This is true That must also be true.
+
+#### When we expect the device under test to PASS or FAIL
+This tells the test runner when to expect Passing/Failing of the device.
+
+```Haskell
+instance IsPlutusModel MyModel MyProperty where
```

Comment:
I think the README is outdated. It should probably be `HasScriptRunner` instead of `IsPlutusModel`.  It is also included in my PR targeting this branch